src/main/java/seedu/multitasky/logic/parser/ListCommandParser.java

```diff
@@ -25,28 +29,81 @@ public ListCommand parse(String args) throws ParseException {
         if (trimmedArgs.isEmpty()) {
             return new ListCommand();
         }
-
-        // Flags are delimited by whitespace
-        final String[] flags = trimmedArgs.split("\\s+");
-        if (!hasValidFlagCombination(flags)) {
+        // Preamble is necessary due to how ArgumentTokenizer works
+        // TODO modify ArgumentTokenizer to be able to detect without preamble
+        ArgumentMultimap argumentMultimap = ArgumentTokenizer.tokenize("preamble " + trimmedArgs,
+                                                                       toPrefixArray(ListCommand.VALID_PREFIXES));
+        ArrayList<String> prefixesPresent = getPresentPrefixes(argumentMultimap, ListCommand.VALID_PREFIXES);
+        if (!hasValidPrefixCombination(prefixesPresent)) {
             throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
                                                    ListCommand.MESSAGE_USAGE));
         }
+        Calendar startDate = getStartDate(argumentMultimap);
+        Calendar endDate = getEndDate(argumentMultimap);
```

Comment:
The Calendar objects can be null, but I just pass it all the way to the DateAndStatusQualifier, which handles the null condition as an actual input type. So there's no where along the way that tries to read it without checking if it's null I think.